Add support for named Elasticsearch client (multi-tenant) (#51422)
quarkusio/quarkus: docs/src/main/asciidoc/elasticsearch-dev-services.adoc

If you want to reuse containers for some Quarkus applications but not all of them,
or some Dev Services but not all of them,
you can disable this feature for a specific Dev Service by setting the configuration property

> xref:elasticsearch-dev-services.adoc#quarkus-elasticsearch-rest-client_quarkus-elasticsearch-devservices-reuse[`quarkus.elasticsearch.devservices.reuse`]

Author: the achhors for config devservices properties changed here... but I'm not sure what exactly contributed to that 😕 , or if maybe it's ok to have the new anchor moving forward ?

Reviewer: Breaking the link is fine here IMO, but the deeper problem is you're bundling the whole ES extension docs in `elasticsearch-dev-services.adoc` instead of just dev service related docs.

to `false`.

== Multiple Elasticsearch service instances

Dev services for Elasticsearch can start multiple instances as necessary.

> Currently, this is only compatible with the `quarkus-elasticsearch-rest-client` extension.

Reviewer: I this true? I see code around multi-tenancy in the Java client support too, and I think the config of that and of the rest client is related. So I'd expect a named Java client to lead to a dev service?

xref:elasticsearch-dev-services.adoc#quarkus-elasticsearch-rest-client_quarkus-elasticsearch-devservices_quarkus-elasticsearch-devservices-reuse[`quarkus.elasticsearch.devservices.reuse`]
to `false`.

== Multiple Elasticsearch service instances

Dev services for Elasticsearch can start multiple instances as necessary.
Currently, this is only compatible with the `quarkus-elasticsearch-rest-client` and
`quarkus-elasticsearch-java-client` extensions.
A separate Dev Service will be started for each named Elasticsearch REST client.

== Current limitations

> Currently, only the default backend for Hibernate Search Elasticsearch is supported.

Reviewer: ```suggestion
Currently, Hibernate Search Elasticsearch can only use dev services for its default backend.
```
